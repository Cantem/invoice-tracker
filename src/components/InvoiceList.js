import React, { Component } from "react";

export default class InvoiceList extends Component {
  render() {
    const sorted = Object.values(this.props.invoices).sort(function(a, b) {
      //sorting the object (invoices) by paid status - paid on top of the list
      return (
        (a.isPaid === null) - (b.isPaid === null) ||
        -(a.isPaid > b.isPaid) ||
        +(a.isPaid < b.isPaid)
      );
    });

    return sorted.map(invoice => (
      <div className="container" key={invoice.id}>
        <div>
          <p> Invoice no.</p>
          {invoice.id}
          /2018
        </div>
        <div>
          <p>Customer:</p>
          {invoice.customer}
        </div>
        <div>
          <p>Value:</p>Â£{invoice.invoiceValue}
        </div>
        <div>
          <p>Payable by:</p>
          {invoice.dueDate}
        </div>
        <div>
          <p>Description:</p>
          {invoice.description}
        </div>
        <div>
          <p>Paid on:</p>
          {invoice.isPaid}
        </div>
        <button onClick={() => this.props.markAsPaid(invoice.id)}>
          Mark as paid
        </button>
      </div>
    ));
  }
}
