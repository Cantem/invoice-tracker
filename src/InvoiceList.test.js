import React from "react";
import renderer from "react-test-renderer";
import InvoiceList from "../src/components/InvoiceList";

it("matches the snapshot", () => {
  const tree = renderer.create(<InvoiceList />);
  expect(tree).toMatchSnapshot();
});
