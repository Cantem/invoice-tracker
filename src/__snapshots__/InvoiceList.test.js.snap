// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="invoice-container"
>
  <div
    className="wrapper"
  >
    <div
      className="invoice"
    >
      <h3>
        Invoice Form
      </h3>
      <img
        alt="invoice"
        className="invoice-image"
        src="invoice.svg"
      />
    </div>
    <div
      className="invoice-structure"
    >
      <form
        onChange={[Function]}
        onSubmit={[Function]}
      >
        <p>
          <label>
            Due date
          </label>
          <input
            className="form-field"
            name="dueDate"
            type="text"
          />
        </p>
        <p>
          <label>
            Invoice value
          </label>
          <input
            className="form-field"
            name="invoiceValue"
            type="text"
          />
        </p>
        <p>
          <label>
            Customer
          </label>
          <input
            className="form-field"
            name="customer"
            type="text"
          />
        </p>
        <p>
          <label>
            Description
          </label>
          <input
            className="form-field"
            name="description"
            rows="5"
            type="text"
          />
        </p>
        <p
          className="full"
        >
          <button
            className="button"
            type="submit"
          >
            Add invoice
          </button>
        </p>
      </form>
    </div>
  </div>
</div>
`;
